stages/org.osbuild.skopeo

```diff
@@ -36,52 +73,73 @@ SCHEMA_2 = r"""
   "required": ["destination"],
   "properties": {
     "destination": {
-      "type": "object",
-      "additionalProperties": false,
-      "required": ["type"],
-      "properties": {
-        "type": {
-          "enum": ["containers-storage"]
-        },
-        "storage-path": {
-          "description": "Container storage location (default /var/lib/containers/storage).",
-          "type": "string"
-        },
-        "storage-driver": {
-          "description": "The container storage driver to use (default overlay).",
-          "type": "string"
-        }
-      }
+      "oneOf": [
+        {"$ref": "#/definitions/destination-containers-storage"},
+        {"$ref": "#/definitions/destination-oci"}
+      ]
     }
   }
 }
 """
 
 
-def main(inputs, output, options):
-    images = containers.parse_containers_input(inputs)
+class ContainersStorageDestination:
+    def __init__(self, storage_path, storage_driver):
+        self.storage_path = storage_path
+        self.storage_driver = storage_driver
 
-    destination = options["destination"]
-    # The destination type is always containers-storage atm, so ignore "type"
+    def skopeo_dest(self, root, image_name):
+        return f"containers-storage:[{self.storage_driver}@{root}{self.storage_path}+/run/containers/storage]{image_name}"
 
-    storage_root = destination.get("storage-path", "/var/lib/containers/storage")
-    storage_driver = destination.get("storage-driver", "overlay")
-
-    for image in images.values():
-        with containers.container_source(image) as (image_name, image_source):
-            dest = f"containers-storage:[{storage_driver}@{output}{storage_root}+/run/containers/storage]{image_name}"
-            subprocess.run(["skopeo", "copy", image_source, dest], check=True)
+    def cleanup(self, root):
+        if self.storage_driver != "overlay":
+            return
 
-    if storage_driver == "overlay":
         # Each time the overlay backend runs on an xfs fs it creates this file:
-        backing_fs_block_dev = os.path.join(output, storage_root.lstrip("/"), "overlay/backingFsBlockDev")
+        backing_fs_block_dev = os.path.join(root, self.storage_path.lstrip("/"), "overlay/backingFsBlockDev")
         # It is not needed in the image as skopeo recreates it each
         # boot, and since it is a block device its existance means we
         # can't store the resulting tree in ostree.
         # Lets just get rid of it.
         if os.path.exists(backing_fs_block_dev):
             os.unlink(backing_fs_block_dev)
 
+
+class OCIDestination:
+    def __init__(self, path):
+        self.path = path
+
+    def skopeo_dest(self, root, image_name):
+        return f"oci:{root}{self.path}"
+
+    def cleanup(self, root):
+        pass
```

Comment:
Defining two classes to differentiate on a formatted string seems a bit excessive, no?

Reply:
I overrefactored! Sorry about that, it's now fixed.